modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/connectivity/oauth/authcode/UpdatingAuthorizationCodeState.java

```diff
@@ -55,14 +66,29 @@ private void update(ResourceOwnerOAuthContext context) {
         onUpdate.accept(context);
       }
     });
+    this.tokenSupplier = () -> {
+      try {
+        return valueOf(dancer.accessToken(getResourceOwnerId()));
+      } catch (RequestAuthenticationException e) {
+        throw new RuntimeException(e);
+      }
+    };
+    this.isClusterEnabled = isClusterEnabled;
   }
 
   @Override
   public String getAccessToken() {
     if (invalidated) {
-      throw new TokenInvalidatedException(
-                                          "OAuth token for resource owner id " + delegate.getResourceOwnerId()
-                                              + " has been invalidated");
+      throw new TokenInvalidatedException(format(INVALIDATED_MESSAGE_TEMPLATE, getResourceOwnerId()));
+    }
+
+    if (isClusterEnabled) {
+      try {
+        tokenSupplier.get();
```

Comment:
i don't understand  this. What this will do is fetch a new access token from the dance. How does this lead to the `invalidated` boolean being updated? At which point is the invalidated boolean distributed to the cluster?